hooks/useCollectionCreate.ts

```diff
@@ -1,11 +1,160 @@
+      return null;
+    }
+
+    setIsCreating(true);
+    setUploadProgress(0);
+
+    try {
+      // Upload image to Arweave
+      const imageUploadResult = await uploadFilesToArweave(
+        null, // previewFile
+        imageFile, // imageFile
+        null, // animationFile
+        "", // existingAnimationUrl
+        setUploadProgress,
+        imageFile.type
+      );
+
+      const imageUri = imageUploadResult.image || imageUploadResult.uploadedImageUri;
+      if (!imageUri) {
+        throw new Error("Failed to upload collection image");
+      }
+
+      // Create collection metadata JSON
+      const metadata = {
+        name,
+        image: imageUri,
+        description: "",
+      };
+
+      // Upload metadata JSON to Arweave
+      setUploadProgress(50);
+      const metadataUri = await uploadJson(metadata);
+      setUploadProgress(75);
+
+      // Call API to create collection
+      const accessToken = await getAccessToken();
+      if (!accessToken) {
+        throw new Error("Authentication required");
+      }
+
+      const result = await callCreateCollectionApi(accessToken, {
+        account: artistWallet,
+        uri: metadataUri,
+        name,
+      });
+      setUploadProgress(100);
+
+      toast.success("Collection created successfully");
+      return {
+        contractAddress: result.contractAddress as Address,
+        metadataUri,
+      };
+    } catch (error: any) {
+      console.error("Error creating collection:", error);
+      toast.error(error?.message || "Failed to create collection");
+      return null;
+    } finally {
+      setIsCreating(false);
+      setUploadProgress(0);
+    }
+  }, [name, imageFile, artistWallet, getAccessToken]);
+
+  const handleSubmit = useCallback(async () => {
+    const result = await createCollection();
+    if (result && artistWallet) {
+      const { contractAddress, metadataUri } = result;
+
+      // Update query cache with the new collection
+      const queryKey = ["collections", artistWallet];
+      queryClient.setQueryData<CollectionsResponse>(queryKey, (oldData) => {
+        if (!oldData) {
+          return {
+            status: "success",
+            collections: [],
+            pagination: { page: 1, limit: 100, total_pages: 1 },
+          };
+        }
+
+        // Create new collection item
+        const newCollection = {
+          id: Date.now().toString(), // Temporary ID, will be replaced on next fetch
+          address: contractAddress,
+          chain_id: CHAIN_ID,
+          uri: metadataUri,
+          name: name.trim(),
+          created_at: new Date().toISOString(),
+          default_admin: {
+            username: null,
+            address: artistWallet,
+          },
+        };
+
+        // Add new collection to the beginning of the array
+        return {
+          ...oldData,
+          collections: [newCollection, ...oldData.collections],
+        };
+      });
```

Comment:
KISS Principle
- actual: manually adding new collection to the dropdown
- required: rely on `indexer` and `tanstack` to `refetch` the list of collections which should include the new collection.